src/Container/HStringContains.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace Haystack\Container;
+
+use Haystack\Helpers\Helper;
+use Haystack\HString;
+
+class HStringContains
+{
+    /** @var string */
+    private $str;
+
+    /** @var string */
+    private $value;
+
+    public function __construct(HString $hString)
+    {
+        $this->str = $hString->toString();
+    }
+
+    /**
+     * @param HString|string $value
+     * @return bool
+     */
+    public function contains($value)
+    {
+        if ($value instanceof HString) {
+            $value = $value->toString();
+        }
+
+        if (is_scalar($value)) {
+            $this->value = (string) $value;
+        } else {
+            throw new \InvalidArgumentException(sprintf("%s is neither a scalar value nor an HString", Helper::getType($value)));
```

Comment:
[NB] This error message says _what_ is wrong but doesn't describe _why_ it's wrong. What's wrong with passing in a value that's not a scalar or HString? What can't be done that was intended to be done when the developer called this method?